src/Mvc/samples/MvcSandbox/HealthChecks/HealthChecksEndpointRouteBuilderExtensions.cs

```diff
@@ -15,7 +15,7 @@ public static class HealthChecksEndpointRouteBuilderExtensions
     {
         private static readonly Random _random = new Random();
 
-        public static IEndpointConventionBuilder MapHealthChecks(this IEndpointRouteBuilder builder, string pattern)
+        public static IEndpointConventions MapHealthChecks(this IEndpointRouteBuilder builder, string pattern)
```

Comment:
`IEndpointConventions` - I strongly object to this name because of the **s** 😞 - where is this defined? I don't see it in the PR

Reply:
It’s not final, I was playing around with names and needed to switch comps so pushed.

To be honest I still don’t like IEndpointConvention. If you look at how the type is used it feels weird. It’s not a convention, the actions passed to it are. Most implementations have a private collection of ‘Action<EndpointBuilder>’ called conventions.

Reply:
ConventionExecutor? What does MVC call the thing that executes IActionResults?

Reply:
The types that implement it do execute the actions on the endpoint as it is being made, but that isn't why this interface is being exposed. It is a hook for a dev to add conventions. One could say, it builds them even.

I took a look around and there is lots of prior art of builders that are used to build up a collection:

https://github.com/aspnet/AspNetCore/blob/master/src/SignalR/src/Microsoft.AspNetCore.SignalR.Common/ISignalRBuilder.cs

https://github.com/aspnet/AspNetCore/blob/master/src/Routing/src/Microsoft.AspNetCore.Routing/IEndpointRouteBuilder.cs

https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/IMvcBuilder.cs

The type could be refactored to be closer to those examples by exposing a collection instead of Apply (which is basically just an Add method).

```cs
public interface IEndpointConventionBuilder
{
    ICollection<Action<EndpointBuilder>> Conventions { get; }
}
```

I don't think there is much value in interacting with the collection beyond adding to it since its content is likely anonymous methods, but implementations of the interface appear to all have the collection anyway, e.g. https://github.com/aspnet/AspNetCore/blob/321327f84b2b22dcff2e9beb06a9a64236c5cced/src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/Builder/DefaultEndpointConventionBuilder.cs

Reply:
If we want to get closer to a collection then how about `void Apply(Action<EndpointBuilder> convention);` -> `void Add(Action<EndpointBuilder> convention);`?

Reply:
I'm 100% ok with:

```cs
public interface IEndpointConventionBuilder
{
    void Add(Action<EndpointBuilder> convention);
}
```